### ISSUE TITLE
unmanaged-cluster minikube provider config enhancements

### ISSUE THREAD

**Issue by jorgemoralespou** [vmware-tanzu/community-edition#3955]
## Feature Request
As a user of the unmanaged clusters with minikube provider, I would love to be able to specify the following minikube configuration:

- vm-driver: seems this is autodetected, an uses hyperkit for macOS (but not sure it works fine)
- cpus. Number of cpus to create the VM with
- memory: Amount of memory to assign to the minikube VM
- disk_size: VM disk size
- cache: To be able to have some images in the minikube cache that will preload automatically into the cluster on creation
- CA. With minikube you can place your own CA in a .minikube folder so that it'll be injected into the machine and automatically added to the OS CAs, so there's easy trust between my machine and my cluster. Related: https://github.com/vmware-tanzu/community-edition/issues/2986

**Comment by joshrosso** [comment]
@jpmcb lets discuss the image resolution on Monday. When I asked to be consistent with the image pre-pull, I totally spaced that the driver choice would dictate the image. We’ll find a simple near-term solution next week.

**Comment by joshrosso** [comment]
Side-note that's related:

We should do some better checking at:

https://github.com/vmware-tanzu/community-edition/blob/3ec336f6bde82ca27cb0eef6bf539f1b4b0d60e3/cli/cmd/plugin/unmanaged-cluster/tkr/tkr.go#L485-L501

Otherwise a malformed TKr may cause us to panic with index out of range.

**Comment by jpmcb** [comment]
> vm-driver: seems this is autodetected, an uses hyperkit for macOS (but not sure it works fine)

Yup, this is minikube's automatic behavior. You can set the driver via: `ProviderConfiguration.driver`

Checkout the doc here (which hasn't been deployed to a version of the documentation yet): https://github.com/vmware-tanzu/community-edition/blob/main/docs/site/content/docs/ref-unmanaged-cluster.md

> cpus. Number of cpus to create the VM with
memory: Amount of memory to assign to the minikube VM
disk_size: VM disk size

Sure, but isn't using `rawMinikubeArgs` sufficient for these uses? Something like:
```yaml
ProviderConfiguration:
  rawMinikubeArgs: --cpus  4 --disk-size 30000mb --memory 4g
``` 
There are [_alot_ of minikube `start` options](https://minikube.sigs.k8s.io/docs/commands/start/). And it would be a bit wild to include all of those in our provider configuration.

But there may be a case made for the most common ones to smooth out the UX for users.

> cache: To be able to have some images in the minikube cache that will preload automatically into the cluster on creation

Which images would be useful for `unmanaged-cluster` to push automatically to the minikube cache that make for a better experience vs just using `minikube cache add`?

> CA. With minikube you can place your own CA in a .minikube folder so that it'll be injected into the machine and automatically added to the OS CAs, so there's easy trust between my machine and my cluster. Related:
https://github.com/vmware-tanzu/community-edition/issues/2986

Can you just use 
```
ProviderConfiguration:
  rawMinikubeArgs: --embed-certs
```
? [Reference docs](https://minikube.sigs.k8s.io/docs/handbook/untrusted_certs/)

**Comment by joshrosso** [comment]
Related, but moving this specific need out of this issue as there are many questions in here. Many of which I believe @jpmcb may have already implemented a solution for.

https://github.com/vmware-tanzu/community-edition/issues/3982

**Comment by jorgemoralespou** [comment]
@jpmcb I can probably resort to minikube raw configs, for many things, yes, but isn't an abstraction meant to simplify the usage of the underlying tool. If, as a user, I need to know all the details of minikube, why would I then use UC and not just minikube?
